crates/credentials/src/presentation_definition.rs

```diff
@@ -236,47 +236,30 @@ impl JsonSchemaBuilder {
 #[cfg(test)]
 mod tests {
     use std::collections::HashSet;
-    use std::fs;
+
+    use test_helpers::TestVectorFile;
 
     use super::PresentationDefinition;
 
     #[derive(Debug, serde::Deserialize)]
-    struct SelectCredentialsVectorInput {
+    struct VectorInput {
         #[serde(rename = "presentationDefinition")]
         pub presentation_definition: PresentationDefinition,
         #[serde(rename = "credentialJwts")]
         pub credential_jwts: Vec<String>,
     }
 
     #[derive(Debug, serde::Deserialize)]
-    struct SelectCredentialsVectorOutput {
+    struct VectorOutput {
         #[serde(rename = "selectedCredentials")]
         pub selected_credentials: Vec<String>,
     }
 
-    #[derive(Debug, serde::Deserialize)]
-    struct SelectCredentialsVector {
-        pub description: String,
-        pub input: SelectCredentialsVectorInput,
-        pub output: SelectCredentialsVectorOutput,
-    }
-
-    #[derive(Debug, serde::Deserialize)]
-    struct Vectors {
-        pub vectors: Vec<SelectCredentialsVector>,
-    }
-
-    fn load_json_fixture(file_path: &str) -> Vectors {
-        let data = fs::read_to_string(file_path).unwrap();
-        let json = serde_json::from_str(&data).unwrap();
-        json
-    }
-
     #[test]
     fn test_web5_spec_test_vectors() {
-        let json_path =
-            "../../web5-spec/test-vectors/presentation_exchange/select_credentials.json";
-        let vectors = load_json_fixture(json_path);
+        let path = "presentation_exchange/select_credentials.json";
+        let vectors: TestVectorFile<VectorInput, VectorOutput> =
+            TestVectorFile::load_from_path(path);
```

Comment:
the typing of the generics is also possible like this:

```suggestion
        let vectors = TestVectorFile::<VectorInput, VectorOutput>::load_from_path(path);
```

both are semantically equivalent, but the type definition for the generics live in different places syntactically

I tend to prefer the proposed way but it's potato, potahto

Reply:
TIL -- I like the latter option too, thanks for showing me!